tests/integration/test_valid_data_in_request.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+# SPDX-License-Identifier: MIT
+
+import time
+
+import utils
+
+
+def test_valid_data_in_request(test_env):
+    """
+    Validate data in the request.
+
+    Process:
+    Send new request to the Cachito API
+    Send request to check status of existing request
+
+    Checks:
+    * Check that response code is 200
+    * Check in the response that state is complete
+    * Check that “packages” and “dependencies” keys have appropriate values
+    """
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"])
+    response_created_req = client.create_new_request(
+        payload={
+            "repo": test_env["get"]["repo"],
+            "ref": test_env["get"]["ref"],
+            "pkg_managers": test_env["get"]["pkg_managers"],
+        },
+    )
+
+    response = client.fetch_request(response_created_req.id)
+    assert response.status == 200
+
+    start_time = time.time()
+    while response.data["state"] == "in_progress":
+        if time.time() - start_time >= 300:
+            raise TimeoutError
+        time.sleep(5)
+        response = client.fetch_request(response_created_req.id)
+    assert response.data["state"] == "complete"
+
+    response_dependencies = list_of_dict_to_list_of_tuples(response.data["dependencies"])
+    expected_dependencies = [tuple(l) for l in test_env["get"]["dependencies"]]
+    assert len(response_dependencies) == len(expected_dependencies)
+    assert response_dependencies == expected_dependencies
+
+    response_packages = list_of_dict_to_list_of_tuples(response.data["packages"])
+    expected_packages = [tuple(l) for l in test_env["get"]["packages"]]
+    assert len(response_packages) == len(expected_packages)
+    assert response_packages == expected_packages
+
+
+def list_of_dict_to_list_of_tuples(data):
+    """
+    Convert the list of dictionaries to the list of tuples.
+
+    :param data: list of dictionaries containing keys name, type and version
+    :return: list of tuples with values name, type and version in this order
+    """
+    sorted_packages = sorted(data, key=lambda x: x["name"])
```

Comment:
Optional: It's preferred to use [operator.itemgetter](https://docs.python.org/3/library/operator.html#operator.itemgetter) instead of created an anonymous function.